pallets/loans/src/types/cashflow.rs

```diff
@@ -173,11 +171,7 @@ impl RepaymentSchedule {
 			.map(|(_, weight)| weight)
 			.try_fold(0, |a, b| a.ensure_add(*b))?;
 
-		let lifetime = maturity.ensure_sub(origination_date)?.ensure_into()?;
-		let interest_rate_per_lifetime = ensure_pow(interest_rate.per_sec()?, lifetime)?;
-		let interest_at_maturity = interest_rate_per_lifetime
-			.ensure_mul_int(principal)?
-			.ensure_sub(principal)?;
+		let interest_at_maturity = current_debt.ensure_sub(principal)?;
```

Comment:
Basically, the principal is different depending on the kind of pricing. The current debt is what previously it was:

```rust
let lifetime = maturity.ensure_sub(origination_date)?.ensure_into()?;
let interest_rate_per_lifetime = ensure_pow(interest_rate.per_sec()?, lifetime)?;
let interest_at_maturity = interest_rate_per_lifetime
	.ensure_mul_int(principal)?
```

We can do this simplification now due it is computed as paid at maturity